Add dependabot, bring all libraries up-to-date, update github actions to java version 2, and add ignores as needed to dependabot

hazendaz opened the pull request:
This brings everything up to date and still java 7 compliant.

davewichers commented on pom.xml:
```diff
@@ -95,25 +100,40 @@
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
```
Why are you adding this library? In the 1.6.5 branch we've dropped adding any logging dependencies entirely, except during test.

hazendaz replied:
```
<groupId>org.apache.httpcomponents</groupId>
<artifactId>httpclient</artifactId>
```

is using commons-logging, I excluded it and added jcl-over-slf4j given this project is using slf4j api so logs are not lost.

davewichers commented on pom.xml:
```diff
@@ -220,12 +240,12 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.0.0-M3</version>
+                <version>3.0.0</version>
```
Did you test this upgrade? When I tried this, I get a parse configuration error during build/test.

hazendaz replied:
1.4+ resolved the issues.

davewichers commented on pom.xml:
```diff
@@ -380,7 +400,7 @@
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
-               <version>3.14.0</version>
+               <version>3.15.0</version>
```
This causes a bunch of errors. I think this has to be left at 3.14.0 until we upgrade to Java 8+.

hazendaz replied:
will look more at this one.

hazendaz replied:
Issue is with ASM.  This may be related to the fact that maven is now suggesting all plugins that use maven core components use provided scope.  In the plugin section, if maven-pmd-plugin is added with asm 9.2, the warnings go away.  Maven pmd plugin is using maven-core 3.1.0 in compile scope with 3.15.0.  It used maven-embedder in provided scope in 3.14.0.  Both versions pick up asm 3.3.1 so based on my testing, 3.3.1 is winning resolution order.  The group id is different between 3.3.1 and 9.2.  This can be shown more directly by using 3.3.1 as it produces same output in that case.  I'm pushing up a small patch here but right place to fix is with pmd.  I had not noticed in my other usage as I typically override pmd to latest which also solves the same issue.

hazendaz replied:
pmd is fixed on master, they moved to maven core 3.1.1 and it doesn't contain asm.  I'm going to submit a patch to them anyways since newer maven is issuing warnings on plugins to not use compile scope.

davewichers commented on pom.xml:
```diff
@@ -173,12 +173,18 @@
         </dependency>
         <dependency>
-          <groupId>org.hamcrest</groupId>
-          <artifactId>hamcrest-core</artifactId>
```
When I try to remove this and run: mvn dependency:analyze, I get this error: [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.2.0:analyze (default-cli) on project antisamy: Cannot analyze dependencies: Trying to force use of dependencies which are not declared: [org.hamcrest:hamcrest-core]

davewichers commented:
@hazendaz - Can you review the pom I have in the 1.6.5 branch. I've included all your proposed changes to the pom in that branch that I think are appropriate/relevant. Can you confirm I've included everything properly?  I haven't done anything with the other 2 files you've proposed changing in this pull request, but those changes look OK to me.

hazendaz commented:
@davewichers Raising new PR to show the diviation.  I noticed 1.6.5 was also not up to date with 'main' so I merged that in this replacement.  I also cleaned up the dependabot add which I think most projects have now (look at apache for example, they practically all are using it and its owned by github).  I also cleaned up the maven workflow update (zulu is the default distribution but in version 2 it is required to be defined - many options now).  For the dependency analyze, that was hard-coded in pom so I updated for the few items that mattered which fixed hamcrest and got jcl replacement of commons logging addressed as well as commons-io addressed so there are no more warnings.

davewichers commented:
Given that we merged the equivalent changes into the 1.6.5 branch, we can just close this right?

hazendaz commented:
yes thanks.
